app/src/main/java/org/mozilla/focus/ext/Uri.kt

```diff
@@ -69,9 +70,11 @@ fun Uri.truncatedPath(): String {
 
 val Uri.isYouTubeDesktop: Boolean
     get() {
-        val host: String? = host // cache it.
-        return host != null &&
-                host.endsWith("youtube.com") &&
+        val host: String? = host // cache properties: their values can change during this call
+        val path: String? = path
+        return if (host == null || path == null) false else {
+            host.endsWith("youtube.com") &&
                 !host.endsWith("m.youtube.com") &&
                 !path.startsWith("/tv")
```

Comment:
Nit: Slightly more verbose, but I think this might be more readable as follows:

```kotlin
        if (host == null || path == null) return false
        return host.endsWith("youtube.com") &&
                !host.endsWith("m.youtube.com") &&
                !path.startsWith("/tv")
```